specs/aset.js

```diff
@@ -37,8 +38,47 @@ function ASet(etype) {
     this.etype = etype;
 }
 
-// TODO: Imeplment set semantics
-
 util.inherits(ASet, AList);
 
+ASet.prototype.uglify = function uglify(list) {
+    var self = this;
+    if (!Array.isArray(list)) {
+        return new Result(SpecError(
+            'ASet::uglify expects an array; received type %s %s val %s',
+            typeof list, list.constructor.name, util.inspect(list)));
+    }
+
+    var tlist = thriftrw.TList(this.etype.typeid);
+    var listUglified = list.map(function uglify(item) {
+       return self.etype.uglify(item);
+    });
+
+    // If there are any errors, return the first.
+    var errors = listUglified.filter(function findErrors(item) {
+        return Boolean(item.err);
+    });
+    if (errors.length) {
+        var error = errors[0].err;
+        error.annotate({
+            type: 'aset'
+        });
+        return new Result(error);
+    }
+
+    var values = listUglified.map(function getValue(item) {
+        return item.value;
+    });
+
+    tlist.elements = uniq(values, function compare(a, b) {
+        var aAsBuffer = bufrw.toBufferResult(thriftrw.getRW(self.etype.typeid), a).value;
```

Comment:
we can probably enforce the key type to be string like the key type of map. do you have a real thrift service which requires set type?